Tools/CMake/AMReXParallelBackends.cmake

```
@@ -116,8 +116,13 @@ if (AMReX_HIP)
    # ARCH flags -- these must be PUBLIC for all downstream targets to use,
    # else there will be a runtime issue (cannot find
    # missing gpu devices)
+   # Equivalently, relocatable-device-code (RDC) flags are needed for `extern`
+   # device variable support (for codes that use global variables on device)
    target_compile_options(amrex
       PUBLIC
-      $<$<COMPILE_LANGUAGE:CXX>:-m64 --amdgpu-target=${AMReX_AMD_ARCH}> )
+      $<$<COMPILE_LANGUAGE:CXX>:-m64 --amdgpu-target=${AMReX_AMD_ARCH} -fgpu-rdc> )
+   target_link_options(amrex
+      PUBLIC
+      $<$<COMPILE_LANGUAGE:CXX>:-fgpu-rdc> )
```

Unconditional to linker?
```suggestion
      -fgpu-rdc)
```